clay.h

```diff
@@ -91,27 +91,14 @@
 
 #define CLAY_STRING(string) CLAY__INIT(Clay_String) { .length = CLAY__STRING_LENGTH(string), .chars = (string) }
 
-static int CLAY__ELEMENT_DEFINITION_LATCH = 0;
```

Comment:
This looks great, only thing is I believe we need to keep this global definition here, because otherwise nesting the macro calls gives us a "declaration shadows local variable" warning for CLAY__ELEMENT_DEFINITION_LATCH 🤔

Reply:
It should be fine to shadow them, and I'm not even sure if it's possible to nest `CLAY` macros, because the nesting would have to take place in the for-loop initialisation. I don't think that you can put a for-loop there.

Reply:
It actually produces an error in my local CLion with clang 🙂
The clay macros expand like this:

```C
CLAY() {
    CLAY() {

    }
}

// Expanded
for (
    uint8_t CLAY__ELEMENT_DEFINITION_LATCH = (Clay__OpenElement(), Clay__ElementPostConfiguration(), 0);
    CLAY__ELEMENT_DEFINITION_LATCH < 1;
    ++CLAY__ELEMENT_DEFINITION_LATCH, Clay__CloseElement()
) {
    for (
    uint8_t CLAY__ELEMENT_DEFINITION_LATCH = (Clay__OpenElement(), Clay__ElementPostConfiguration(), 0);
    CLAY__ELEMENT_DEFINITION_LATCH < 1;
    ++CLAY__ELEMENT_DEFINITION_LATCH, Clay__CloseElement()
    ) {
        // contents
    }
}
```

Reply:
Oh hmm, I don't any warnings for that at all, with clang or gcc with -Wall -Wextra -Wpedantic. What warning is it exactly?

Reply:
<img width="421" alt="Screenshot 2024-12-29 at 2 24 26 pm" src="https://github.com/user-attachments/assets/178e948c-7b37-4a7a-9835-1b4b5c53067f" />

Reply:
Ah ok, I'll put it back to being static then :-)

Reply:
It seems to be under `-Wshadow` 🤔

Reply:
Hmm, even manually adding that flag doesn't seem to get it to show me that warning. Anyway, it does seem better to just keep it static if there's a chance it'll get someone's IDE to complain.